### PR TITLE
Esrb filtering

### PR DESCRIPTION
**Pull request by pjthomso**

### CODE REVIEW

**Comment by BlackthornYugen** [comment]
+1

File: SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs [hunk 1]
```diff
@@ -1,150 +1,218 @@
...
+                new ESRB("AO", "Adult Only", 18)
+            };
+        
+        #region Multiple Roles
+
+        /// <summary>
+        /// Employee side - list all games
+        /// Member side - search feature, list matching games
+        /// </summary>
+        /// <returns>list of games view</returns>
```
Comment (ndahlquist-cc) [inline SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs]: the <returns> content needs updating?

File: SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs [hunk 2]
```diff
@@ -1,150 +1,218 @@
...
-        /// </summary>
-        /// <returns>List of games view</returns>
-        public ActionResult SearchGames(string nameSearch, int[] platformSearch, int[] categorySearch, string[] esrbSearch, bool isInclusive = false)
-        {           
-            IEnumerable<Game> gameList = db.Games.Include(g => g.Platform).Include(g => g.Categories);
-            //Name search query
-            if(nameSearch != null)
-            {
+
+        /// <summary>
```
Comment (ndahlquist-cc) [inline SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs:75]: needs params in the doc comments?

File: SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs [hunk 3]
```diff
@@ -158,207 +226,207 @@ public ActionResult GameManagement()
...
+        }
+
+        /// <summary>
+        /// Employee Side - post back for game creation, attempt to save to db
+        /// </summary>
+        /// <param name="game">game object</param>
+        /// <returns>view of games' list</returns>
+        //ADD EMPLOYY AUTH
+        [HttpPost]
         [ValidateAntiForgeryToken]
```
Comment (ndahlquist-cc) [inline SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs:255]: params should include platform/category? remove note to add employee auth

File: SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs [hunk 4]
```diff
@@ -158,207 +226,207 @@ public ActionResult GameManagement()
...
+
+            Create();
+            return View(game);
+        }
+
+        /// <summary>
+        /// Employee side - edit game
+        /// </summary>
+        /// <param name="id">game id</param>
         /// <returns>add/edit game view</returns>
```
Comment (ndahlquist-cc) [inline SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs]: should only be Edit game view

File: SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs [hunk 5]
```diff
@@ -158,207 +226,207 @@ public ActionResult GameManagement()
...
+            PopulateDropdownData();
+            return View(game);
+        }
+
+        /// <summary>
+        /// Employee side - post back for edit game, save to db
+        /// </summary>
+        /// <param name="game">game object</param>
+        /// <returns>list of games view</returns>
+        [HttpPost]
```
Comment (ndahlquist-cc) [inline SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/GameController.cs:335]: add category param

**Comment by BlackthornYugen** [comment]
@pjthomso My +1 still stands and since all of the feedback was on doc comments this should be good to merge right? 

When you have time could you merge master into your branch, make sure it still builds and then merge to master?

**Comment by pjthomso** [comment]
updated from master and fixed the doc comments, It builds and runs but i'm at school and having uses updating the database using package manager so I wasn't able to double check the pages which had conflicts to ensure they still looked proper.

**Comment by pjthomso** [comment]
Updated from master again -.-

**Pull request merged**